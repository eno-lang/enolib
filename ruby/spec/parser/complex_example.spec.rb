describe Enolib::Parser do
  context 'with a complex sample' do
    input = File.read(File.join(__dir__, 'samples/complex.eno'))

    context = Enolib::Context.new(input)

    it 'parses as expected' do
      expect(context.document).to match_snapshot
    end
  end
end
