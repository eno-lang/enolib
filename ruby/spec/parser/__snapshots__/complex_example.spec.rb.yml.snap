---
"./spec/parser/complex_example.spec.rb[1:1:1]": &1
  :elements:
  - :index: 19
    :line: 2
    :ranges:
      :line:
      - 19
      - 57
      :element_operator:
      - 23
      - 24
      :key:
      - 19
      - 23
      :value:
      - 25
      - 57
    :key: Date
    :type: :field
    :value: 1st of November in the year 2017
    :parent: *1
  - :index: 59
    :line: 4
    :ranges:
      :line:
      - 59
      - 78
      :multiline_field_operator:
      - 59
      - 61
      :key:
      - 62
      - 78
    :key: Book Description
    :parent: *1
    :type: :multiline_field_begin
    :lines:
    - :line: 5
      :ranges:
        :line:
        - 79
        - 98
        :value:
        - 79
        - 98
      :type: :multiline_field_value
    - :line: 6
      :ranges:
        :line:
        - 99
        - 108
        :value:
        - 99
        - 108
      :type: :multiline_field_value
    :end:
      :length: 128
      :line: 7
      :ranges:
        :key:
        - 112
        - 128
        :line:
        - 109
        - 128
        :multiline_field_operator:
        - 109
        - 111
      :type: :multiline_field_end
  - &2
    :index: 130
    :line: 9
    :ranges:
      :line:
      - 130
      - 144
      :element_operator:
      - 143
      - 144
      :key:
      - 130
      - 143
    :key: Shopping List
    :type: :list
    :parent: *1
    :items:
    - :index: 145
      :line: 10
      :ranges:
        :line:
        - 145
        - 153
        :item_operator:
        - 145
        - 146
        :value:
        - 147
        - 153
      :type: :list_item
      :value: Apples
      :parent: *2
    - :index: 154
      :line: 11
      :ranges:
        :line:
        - 154
        - 163
        :item_operator:
        - 154
        - 155
        :value:
        - 156
        - 163
      :type: :list_item
      :value: Oranges
      :parent: *2
  - &3
    :index: 165
    :line: 13
    :ranges:
      :line:
      - 165
      - 183
      :element_operator:
      - 182
      - 183
      :key:
      - 165
      - 182
    :key: Telephone Numbers
    :type: :fieldset
    :parent: *1
    :entries:
    - :index: 184
      :line: 14
      :ranges:
        :line:
        - 184
        - 207
        :key:
        - 184
        - 187
        :entry_operator:
        - 188
        - 189
        :value:
        - 190
        - 207
      :type: :fieldset_entry
      :key: Ben
      :value: "+49 1943 24724784"
      :parent: *3
    - :index: 208
      :line: 15
      :ranges:
        :line:
        - 208
        - 234
        :key:
        - 208
        - 214
        :entry_operator:
        - 215
        - 216
        :value:
        - 217
        - 234
      :type: :fieldset_entry
      :key: Rachel
      :value: "+59 3459 35935593"
      :parent: *3
    - :index: 235
      :line: 16
      :ranges:
        :line:
        - 235
        - 261
        :entry_operator:
        - 242
        - 243
        :escape_begin_operator:
        - 235
        - 236
        :escape_end_operator:
        - 240
        - 241
        :key:
        - 236
        - 240
        :value:
        - 244
        - 261
      :type: :fieldset_entry
      :key: Dave
      :value: "+69 9025 93593531"
      :parent: *3
  - &8
    :index: 263
    :line: 18
    :ranges:
      :line:
      - 263
      - 269
      :section_operator:
      - 263
      - 264
      :key:
      - 265
      - 269
    :elements:
    - &7
      :index: 270
      :line: 19
      :ranges:
        :line:
        - 270
        - 278
        :section_operator:
        - 270
        - 272
        :key:
        - 273
        - 278
      :elements:
      - &4
        :index: 279
        :line: 20
        :ranges:
          :line:
          - 279
          - 291
          :section_operator:
          - 279
          - 282
          :key:
          - 283
          - 291
        :elements:
        - :index: 293
          :line: 22
          :ranges:
            :line:
            - 293
            - 336
            :element_operator:
            - 310
            - 311
            :escape_begin_operator:
            - 293
            - 294
            :escape_end_operator:
            - 309
            - 310
            :key:
            - 294
            - 309
            :value:
            - 312
            - 336
          :key: "--format pretty"
          :type: :field
          :value: Pretty formatting option
          :parent: *4
        - &5
          :index: 338
          :line: 24
          :ranges:
            :line:
            - 338
            - 350
            :element_operator:
            - 349
            - 350
            :escape_begin_operator:
            - 338
            - 339
            :escape_end_operator:
            - 348
            - 349
            :key:
            - 339
            - 348
          :key: "> Friends"
          :type: :list
          :parent: *4
          :items:
          - :index: 351
            :line: 25
            :ranges:
              :line:
              - 351
              - 368
              :item_operator:
              - 351
              - 352
              :value:
              - 353
              - 368
            :type: :list_item
            :value: Jack the Ripper
            :parent: *5
          - :index: 369
            :line: 26
            :ranges:
              :line:
              - 369
              - 383
              :item_operator:
              - 369
              - 370
              :value:
              - 371
              - 383
            :type: :list_item
            :value: Frankenstein
            :parent: *5
        - &6
          :index: 385
          :line: 28
          :ranges:
            :line:
            - 385
            - 416
            :element_operator:
            - 415
            - 416
            :escape_begin_operator:
            - 385
            - 387
            :escape_end_operator:
            - 413
            - 415
            :key:
            - 387
            - 413
          :key: Use `${arg}` style options
          :type: :fieldset
          :parent: *4
          :entries:
          - :index: 417
            :line: 29
            :ranges:
              :line:
              - 417
              - 470
              :key:
              - 417
              - 431
              :entry_operator:
              - 432
              - 433
              :value:
              - 434
              - 470
            :type: :fieldset_entry
            :key: "'--output dir'"
            :value: Specify a different output directory
            :parent: *6
          - :index: 471
            :line: 30
            :ranges:
              :line:
              - 471
              - 516
              :key:
              - 471
              - 491
              :entry_operator:
              - 492
              - 493
              :value:
              - 494
              - 516
            :type: :fieldset_entry
            :key: "'--filters disabled'"
            :value: Enable/disable filters
            :parent: *6
        - :index: 518
          :line: 32
          :ranges:
            :line:
            - 518
            - 520
            :element_operator:
            - 519
            - 520
            :key:
            - 518
            - 519
          :key: I
          :type: :field
          :parent: *4
          :continuations:
          - :index: 521
            :line: 33
            :ranges:
              :line:
              - 521
              - 522
              :direct_line_continuation_operator:
              - 521
              - 522
            :type: :continuation
          - :index: 523
            :line: 34
            :ranges:
              :line:
              - 523
              - 531
              :spaced_line_continuation_operator:
              - 523
              - 524
              :value:
              - 525
              - 531
            :spaced: true
            :type: :continuation
            :value: append
          - :index: 532
            :line: 35
            :ranges:
              :line:
              - 532
              - 533
              :spaced_line_continuation_operator:
              - 532
              - 533
            :spaced: true
            :type: :continuation
          - :index: 534
            :line: 36
            :ranges:
              :line:
              - 534
              - 539
              :direct_line_continuation_operator:
              - 534
              - 535
              :value:
              - 536
              - 539
            :type: :continuation
            :value: all
          - :index: 540
            :line: 37
            :ranges:
              :line:
              - 540
              - 545
              :direct_line_continuation_operator:
              - 540
              - 541
              :value:
              - 542
              - 545
            :type: :continuation
            :value: the
          - :index: 546
            :line: 38
            :ranges:
              :line:
              - 546
              - 554
              :spaced_line_continuation_operator:
              - 546
              - 547
              :value:
              - 548
              - 554
            :spaced: true
            :type: :continuation
            :value: things
        :type: :section
        :key: Left Arm
        :parent: *7
      :type: :section
      :key: Limbs
      :parent: *8
    :type: :section
    :key: Body
    :parent: *1
  :type: :document
